src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs

```diff
@@ -99,46 +157,57 @@ public virtual async Task<Response<ConfigurationSetting>> AddAsync(Configuration
         /// <param name="value">The value of the configuration setting.</param>
         /// <param name="label">The value used to group configuration settings.</param>
         /// <param name="cancellation">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        public virtual async Task<Response<ConfigurationSetting>> AddAsync(string key, string value, string label = default, CancellationToken cancellation = default)
+        public virtual Response<ConfigurationSetting> Set(ConfigurationSetting setting, CancellationToken cancellationToken = default)
         {
-            if (string.IsNullOrEmpty(key)) throw new ArgumentNullException($"{nameof(key)}");
-            return await AddAsync(new ConfigurationSetting(key, value, label), cancellation);
-        }
+            using (Request request = CreateSetRequest(setting))
+            {
+                var response = _pipeline.SendRequest(request, cancellationToken);
 
-        /// <summary>
+                switch (response.Status)
```

Comment:
Could we also have something like `CreateSetResponse` method so that we don't repeat this section for both async and sync methods?

Reply:
Problem is that there are sync and async versions of that method so they are unfortunately non-unifiable.